packages/node/src/executors/package/package.impl.ts

```diff
@@ -43,9 +44,53 @@ export async function packageExecutor(
     dependencies
   );
 
-  await copyAssetFiles(normalizedOptions.files);
+  // Since we delegate watch mode to tsc, we need to repopulate assets after each build cycle.
+  if ('subscribe' in result) {
+    let lock = false;
+    result
+      .pipe(
+        debounceTime(150),
+        filter(() => !lock)
+      )
+      .subscribe(async () => {
+        lock = true;
+        await updatePackageAndCopyAssets(
+          normalizedOptions,
+          context,
+          target,
+          dependencies
+        );
+        lock = false;
+      });
+    await new Promise<any>(() => {});
```

Comment:
You could return the observable `.toPromise()` with a `tap`

Reply:
No more observables. Uses callback now.